uktrade/data-hub-frontend
# Display interactions linked to an opportunity

## Description of change

**What the PR does**

This PR adds tab navigation to the page which displays an individual large capital investment opportunity, and in the new `Interactions` tab, displays the interactions linked to the opportunity.

**Implementation details**

- As the tab navigation wraps the content of the opportunity page, I have moved the task which gets the opportunity details to the tabnav component, as well as mounting the tab nav component rather than the opportunities component.
- I have used the react-router `<Route>` component to give certain components access to the `location` and `history` props, rather than connecting the components to this slice of state.
- I have renamed some components related to opportunities, as with the introduction of the tabs, calling the opportunity details table `OpportunityDetails` no longer worked.

**To note**
- The commit `Improve the Resource abstraction`, has been cherry-picked from [this PR](https://github.com/uktrade/data-hub-frontend/pull/3986), and can be removed once that PR is merged.
- Don't be alarmed by the number of changes! Well over 1000 lines are from new fixtures, and more from the aforementioned commit.

## Test instructions

- On the heroku deployment of this branch go to [this opportunity](https://data-hub-fro-feature-di-ccdmez.herokuapp.com/investments/opportunities/503e38aa-d2ba-46f1-9ff6-0dca44b822c1/details)
- Click on the interactions tab
- You should see a list of interactions linked to this opportunity
- To see the pagination there needs to be more than 10 interactions against an opportunity, you can set the `large_capital_opportunity` field on interactions in django admin to see this

## Screenshots
### Before

![Screenshot 2021-09-22 at 12 04 21](https://user-images.githubusercontent.com/22460823/134332670-91ef5366-92a8-4656-af43-ea842a1380d6.png)


### After

![Screenshot 2021-09-22 at 12 04 48](https://user-images.githubusercontent.com/22460823/134332731-85078071-6e91-4d99-a3cc-a8b6d669b98f.png)


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
